File: src/main/java/de/maggiwuerze/xdccloader/persistence/DatabaseLoader.java
package de.maggiwuerze.xdccloader.persistence;

import de.maggiwuerze.xdccloader.model.entity.Bot;
import de.maggiwuerze.xdccloader.model.entity.Channel;
import de.maggiwuerze.xdccloader.model.entity.Server;
import de.maggiwuerze.xdccloader.model.entity.UserSettings;
import jakarta.transaction.Transactional;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.CommandLineRunner;
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
import org.springframework.stereotype.Component;


@Component
class DatabaseLoader implements CommandLineRunner {


	@Autowired
	SettingsRepository settingsRepository;
	@Autowired
	ChannelRepository channelRepository;
	@Autowired
	ServerRepository serverRepository;
	@Autowired
	IrcUserRepository ircUserRepository;
	@Autowired
	UserSettingsRepository userSettingsRepository;

	@Transactional
	@Override
	public void run(String... strings) {
		UserSettings userSettings = userSettingsRepository.save(new UserSettings());
		this.settingsRepository.save(userSettings);

		Server server = serverRepository.save(new Server("Rizon", "irc.rizon.net"));
		Channel channel = channelRepository.save(new Channel("#subsplease"));
		Bot bot = this.ircUserRepository.save(new Bot(server, channel, "Ginpachi-Sensei", "xdcc send #%s", 3L));

	}
}

File: src/main/kotlin/de/maggiwuerze/xdccwebloader/persistence/DatabaseLoader.kt
package de.maggiwuerze.xdccwebloader.persistence

import de.maggiwuerze.xdccwebloader.persistence.entity.Bot
import de.maggiwuerze.xdccwebloader.persistence.entity.Channel
import de.maggiwuerze.xdccwebloader.persistence.entity.Server
import de.maggiwuerze.xdccwebloader.persistence.entity.UserSettings
import jakarta.transaction.Transactional
import org.springframework.boot.CommandLineRunner
import org.springframework.stereotype.Component

@Component
class DatabaseLoader(

    var settingsRepository: UserSettingsRepository,

    var channelRepository: ChannelRepository,

    var serverRepository: ServerRepository,

    var ircUserRepository: IrcUserRepository,

    var userSettingsRepository: UserSettingsRepository,
) : CommandLineRunner {

    @Transactional
    override fun run(vararg strings: String) {
        val userSettings: UserSettings = userSettingsRepository.save(UserSettings())
        this.settingsRepository.save(userSettings)

        val server: Server = serverRepository.save(Server(name = "Abjects", serverUrl = "irc.abjects.net"))
        val channel: Channel = channelRepository.save(Channel(name = "#beast-xdcc"))
        this.ircUserRepository.save(
            Bot(
                server = server,
                channel = channel,
                name = "BEAST-X-CHEDDAR",
                pattern = "xdcc send #%s",
                maxParallelDownloads = 3L
            )
        )
        this.ircUserRepository.save(
            Bot(
                server = server,
                channel = channel,
                name = "BEAST-X-REDRUNTZ",
                pattern = "xdcc send #%s",
                maxParallelDownloads = 3L
            )
        )
    }
}